File: passing_cars.cpp
// File: passing_cars.cpp
// Author: Thomas Wood -- thomas@synpon.com

int solution(vector<int> &A) {
    // write your code in C++14 (g++ 6.2.0)
    int n = A.size();
    vector<int> scan;

    int s = 0;
    for (int j=0; j < n; j++) {
        scan.push_back(s);
        s += A[j];
    }

    unsigned int passing = 0;
    for (int j=0; j < n; j++) {
        if (A[j] == 0) {
            passing += s - scan[j];
        }
    }
    if (passing <= 1000000000) {
        return passing;
    }
    else {
        return -1;
    }
}

File: passing_cars.py
#! /usr/bin/env/ python
# -*- coding: utf-8

def solution(A):
    # write your code in Python 2.7
    scan = []
    s = 0
    if len(A) < 2:
        return 0
    for x in A:
        s += x
        scan.append(s)

    total = scan[-1]

    passing = 0
    for k, x in enumerate(scan):
        if A[k] == 0:
            passing += total - x


    if passing <= 1e9:
        return passing
    else:
        return -1
